src/core/model/table/dynamic_table_data.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "config/exceptions.h"
+#include "config/tabular_data/input_table_type.h"
+
+namespace model {
+
+struct DynamicTableData {
+private:
+    std::vector<std::vector<std::string>> columns_;
+    std::unordered_set<size_t> deleted_rows_{};
+
+public:
+    DynamicTableData(config::InputTable input_table) {
```

Comment:
This function doesn't take ownership of the `shared_ptr`, use `IDatasetStream&` instead.